crates/chia-datalayer/src/merkle/test_util.rs

```diff
@@ -0,0 +1,84 @@
+use crate::merkle::dot::DotLines;
+use crate::merkle::util::{sha256_bytes, sha256_num};
+use crate::{Hash, InsertLocation, KeyId, MerkleBlob, Side, TreeIndex, ValueId};
+use chia_protocol::Bytes32;
+use rstest::fixture;
+
+pub const HASH_ZERO: Hash = Hash(Bytes32::new([0; 32]));
+pub const HASH_ONE: Hash = Hash(Bytes32::new([1; 32]));
+pub const HASH_TWO: Hash = Hash(Bytes32::new([2; 32]));
+
+pub fn open_dot(_lines: &mut DotLines) {
+    // crate::merkle::dot::open_dot(_lines);
+}
+
+#[fixture]
+pub fn small_blob() -> MerkleBlob {
+    let mut blob = MerkleBlob::new(vec![]).unwrap();
+
+    blob.insert(
+        KeyId(0x0001_0203_0405_0607),
+        ValueId(0x1011_1213_1415_1617),
+        &sha256_num(0x1020),
+        InsertLocation::Auto {},
+    )
+    .unwrap();
+
+    blob.insert(
+        KeyId(0x2021_2223_2425_2627),
+        ValueId(0x3031_3233_3435_3637),
+        &sha256_num(0x2030),
+        InsertLocation::Auto {},
+    )
+    .unwrap();
+
+    blob
+}
+
+#[fixture]
+pub fn traversal_blob(mut small_blob: MerkleBlob) -> MerkleBlob {
+    small_blob
+        .insert(
+            KeyId(103),
+            ValueId(204),
+            &sha256_num(0x1324),
+            InsertLocation::Leaf {
+                index: TreeIndex(1),
+                side: Side::Right,
+            },
+        )
+        .unwrap();
+    small_blob
+        .insert(
+            KeyId(307),
+            ValueId(404),
+            &sha256_num(0x9183),
+            InsertLocation::Leaf {
+                index: TreeIndex(3),
+                side: Side::Right,
+            },
+        )
+        .unwrap();
+
+    small_blob.calculate_lazy_hashes().unwrap();
+    small_blob
+}
+
+pub fn generate_kvid(seed: i64) -> (KeyId, ValueId) {
+    let mut kv_ids: Vec<i64> = Vec::new();
+
+    for offset in 0..2 {
+        let seed_int = 2i64 * seed + offset;
```

Comment:
if `seed` is allowed to be a sufficiently large or sufficiently small value, this may overflow the integer. If that's deliberate, you should use `seed.wrapping_mul(2).wrapping_add(offset)`

Reply:
looks like we're trying to spread the seed space out so we can get two unique results from one seed.  the key and the value.  i'll constrain the seed parameter to i32.

- [ ] https://github.com/Chia-Network/chia_rs/pull/1214